app/access/api.py

```diff
@@ -75,3 +80,32 @@ def create(request: HttpRequest, user_in: UserSchema) -> UserSchema:
         if not created:
             raise HttpError(500, "Internal Server Error")
     return user_to_response(user_out)
+
+
+@router.put("users/{username}")
+def update_user(request: HttpRequest, username: str, user_in: UserSchema) -> HttpResponse:
+    """Update the given user with the given user data.
+
+    Return HTTP status code
+    - 200 (OK) if the user has been updated successfully
+    - 404 (Not Found) if there is no user with the given username
+    - 500 (Internal Server Error) if there is an inconsistency with Cognito
+    - 500 (Internal Server Error) if there no provider with the given ID
+    - 503 (Service Unavailable) if Cognito cannot be reached
+    """
+    with transaction.atomic():
+        user_object = User.objects.select_for_update().filter(username=username).first()
+        if not user_object:
+            raise Http404()
+
+        if not Provider.objects.filter(id=user_in.provider_id).exists():
+            raise HttpError(HTTPStatus.INTERNAL_SERVER_ERROR, "Provider does not exist")
```

Comment:
I would use some sort of 4xx response since this is under the control of the user of the API.

Reply:
Good point, will change to 404 :wrench:

Reply:
Sorry for nitpicky, I wonder if this should be 403 Bad Request, as 404 would point to the resource that is being updated itself, i.e. the user here.

Reply:
I think `422` is mostly used for webdav so I'd probably use `400 Bad Request`.